### PR TITLE
Updated all package dependencies to MSAL v4 and refactored accordingly, plus see notes.

### PR DESCRIPTION
**Pull request by AndyMason75**
Switched session cache to memory cache
Added post logout redirect option
Added compatability with WebForms
Updated documentation

### CODE REVIEW

File: Veracity.Authentication.OpenIDConnect.AspNet/Startup.cs [hunk 1]
```diff
@@ -41,47 +51,56 @@ private void ConfigureAuth(IAppBuilder app)
...
                     Notifications = new OpenIdConnectAuthenticationNotifications
                     {
                         AuthorizationCodeReceived = this.OnAuthorizationCodeReceivedAsync,
-                        AuthenticationFailed = this.AuthenticationFailed
+                        AuthenticationFailed = this.OnAuthenticationFailed
                     },
                     ResponseType = OpenIdConnectResponseType.CodeIdToken,
                     TokenValidationParameters = new TokenValidationParameters
                     {
+                        ValidateIssuer = false,
```
Comment (mingderekwang) [inline Veracity.Authentication.OpenIDConnect.AspNet/Startup.cs]: The issuer should be validated.

File: Veracity.Authentication.OpenIDConnect.AspNet/MSALPerUserMemoryTokenCache.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    /// <seealso cref="https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/token-cache-serialization"/>
+    public class MSALPerUserMemoryTokenCache
+    {
+        /// <summary>
+        /// The backing MemoryCache instance
+        /// </summary>
+        private readonly MemoryCache memoryCache = MemoryCache.Default;
+
+        /// <summary>
+        /// The duration till the tokens are kept in memory cache. In production, a higher value, upto 90 days is recommended.
```
Comment (mingderekwang) [inline Veracity.Authentication.OpenIDConnect.AspNet/MSALPerUserMemoryTokenCache.cs]: The lifetime of refresh token is one week in Veracity

Comment (mingderekwang) [inline Veracity.Authentication.OpenIDConnect.AspNet/MSALPerUserMemoryTokenCache.cs]: We recommended that the cache lifetime should be the half of refresh token  life time.

File: Veracity.Authentication.OpenIDConnect.AspNet/Startup.cs (hunk 1 above)
Comment (AndyMason75) [reply Veracity.Authentication.OpenIDConnect.AspNet/Startup.cs]: Fixed in a later check-in

File: Veracity.Authentication.OpenIDConnect.AspNet/MSALPerUserMemoryTokenCache.cs (hunk 2 above)
Comment (AndyMason75) [reply Veracity.Authentication.OpenIDConnect.AspNet/MSALPerUserMemoryTokenCache.cs]: Fixed in a later check-in

Comment (AndyMason75) [reply Veracity.Authentication.OpenIDConnect.AspNet/MSALPerUserMemoryTokenCache.cs]: Fixed in a later check-in

**Comment by AndyMason75** [comment]
Hope this is OK

**Pull request merged**